fsspec/core.py

```diff
@@ -706,7 +706,7 @@ def _expand_paths(path, name_function, num):
         assert len(path) == num
         paths = list(path)
     else:
-        raise ValueError(
+        raise TypeError(
```

Comment:
This is a breaking change. I have committed it for documentation purposes, but perhaps I should discard the relevant commit.

Reply:
I don't see a big reason to change it. Both say "you passed the wrong thing" in this context.

Reply:
`TypeError` is for arguments of the wrong type, while `ValueError` is for argument with a wrong value.

Reply:
> TypeError is for arguments of the wrong type, while ValueError is for argument with a wrong value.

I know that's technically the case, but here that can be several types; let's not change it.

Reply:
How is the number of possible types relevant?

Any way, I have removed that commit.